yb-voyager/src/srcdb/yugabytedb.go

```diff
@@ -767,13 +767,48 @@ WHERE parent.relname='%s' AND nmsp_parent.nspname = '%s' `, tname, sname)
 }
 
 const ybQueryTmplForUniqCols = `
-SELECT tc.table_schema, tc.table_name, kcu.column_name
-FROM information_schema.table_constraints tc
-JOIN information_schema.key_column_usage kcu
-    ON tc.constraint_name = kcu.constraint_name
-	AND tc.table_schema = kcu.table_schema
-    AND tc.table_name = kcu.table_name
-WHERE tc.table_schema = ANY('{%s}') AND tc.table_name = ANY('{%s}') AND tc.constraint_type = 'UNIQUE';
+WITH unique_constraints AS (
+    SELECT
+        tc.table_schema,
+        tc.table_name,
+        kcu.column_name
+    FROM
+        information_schema.table_constraints tc
+    JOIN
+        information_schema.key_column_usage kcu
+        ON tc.constraint_name = kcu.constraint_name
+        AND tc.table_schema = kcu.table_schema
+        AND tc.table_name = kcu.table_name
+    WHERE
+        tc.constraint_type = 'UNIQUE'
+        AND tc.table_schema = ANY('{%s}')
+        AND tc.table_name = ANY('{%s}')
+),
+unique_indexes AS (
+    SELECT
+        n.nspname AS table_schema,
+        t.relname AS table_name,
+        a.attname AS column_name
+    FROM
+        pg_index ix
+    JOIN
+        pg_class t ON t.oid = ix.indrelid
+    JOIN
+        pg_namespace n ON n.oid = t.relnamespace
+    JOIN
+        pg_attribute a ON a.attrelid = t.oid AND a.attnum = ANY(ix.indkey)
```

Comment:
`AND a.attnum = ANY(ix.indkey)` what does this mean? Can you pls add a comment?

Reply:
done.
It is basically matching the column number which are part of index with that number in pg_attribute table

Reply:
Oh. Then why are we doing `ANY()` ? Shouldn't `a.attnum = ix.indkey` work?